### DISCUSSION TITLE
How can I not accept all topics? ( # topic )

### DISCUSSION THREAD

**Discussion by bahientn9x** [emqx/emqx#6196] [Q&A]
Help!
Currently i am facing an attacker on the server by listening to all topic (# topic or multiple levels) and repeating the packet causing iot devices to misbehave. I want the topic to be able to send and receive only one destination. So what should I do with the ACL ? 
Thanks a lot

**Comment by zmstone** [comment]
Hi @bahientn9x which version are you running?
1. Please disable anonymous `allow_anonymous = false` in emqx.conf
2. ACL file is the most effective and lightweight solution: https://docs.emqx.io/en/broker/v4.3/advanced/acl-file.html

EDIT:
to deny multiple pub/sub against many layers, use a rule with a pattern `+/+.../#` e.g.
`{deny, all, pubsub, ["+/+/+/+/#"]}` should deny both publish and subscribe requests against topics more than 4 levels.

**Reply by bahientn9x** [reply]
The version i am using is 4.3.5. I turned off the anonymous function and used ACL but I can't filter out all the wildcards.
My topic has 7 levels and hacker can use + and # characters for multiple levels. So I want to stop using wildcards

**Reply by zmstone** [reply]
block their IP ?

**Reply by bahientn9x** [reply]
i tried but he changed the ip.

**Reply by zmstone** [reply]
or enable authentication, that means your clients will need to start authenticating themselves

**Reply by zmstone** [reply]
or whitelisting your IPs

**Reply by bahientn9x** [reply]
he got the account and password authenticated by a system flaw. And I can't change that authentication because my devices are using them.

**Reply by bahientn9x** [reply]
I have a lot of customers so I have a lot of IPs so I can't use a firewall to filter IPs

**Reply by bahientn9x** [reply]
So the only way I can think of is to turn off the wildcard function so that he can't catch other packets to repeat them.

**Reply by zmstone** [reply]
but there is no way to stop them from subscribing to non-wildcard topics ?

**Reply by zmstone** [reply]
if they are packet snifferring, I guess the only way forward is to change password, and start using TLS (should be enough if terminate by a load balancer).
